internal/reminder/reminder.go

```diff
@@ -149,70 +137,77 @@ func (r *reminder) Stop() {
+	for _, repo := range repos {
+		logger.Debug().Msgf("updating reminder_last_sent for repository: %s", repo.ID)
+		logger.Debug().Msgf("previous reminder_last_sent: %s", repo.ReminderLastSent.Time)
+		err := r.store.UpdateReminderLastSentById(ctx, repo.ID)
+		if err != nil {
+			logger.Error().Err(err).Msgf("unable to update reminder_last_sent for repository: %s", repo.ID)
+			return []error{err}
+		}
+		// TODO: Send the actual reminders
 	}
 
-	return os.WriteFile(r.cfg.CursorFile, buf.Bytes(), 0600)
+	return nil
 }
 
-// restoreCursorState restores the cursor state from a file
-// Not thread-safe, should be called from a single goroutine
-func (r *reminder) restoreCursorState(ctx context.Context) error {
+func (r *reminder) getRepositoryBatch(ctx context.Context) ([]db.Repository, error) {
 	logger := zerolog.Ctx(ctx)
-	logger.Debug().Msg("restoring cursor state")
-
-	if _, err := os.Stat(r.cfg.CursorFile); os.IsNotExist(err) {
-		return nil
-	}
 
-	data, err := os.ReadFile(r.cfg.CursorFile)
+	logger.Debug().Msgf("fetching repositories after cursor: %s", r.repositoryCursor)
+	repos, err := r.store.ListEligibleRepositoriesAfterID(ctx, db.ListEligibleRepositoriesAfterIDParams{
+		ID:                r.repositoryCursor,
+		Limit:             int64(r.cfg.RecurrenceConfig.BatchSize),
+		MinElapsedSeconds: r.cfg.RecurrenceConfig.MinElapsed.Seconds(),
+	})
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	buf := bytes.NewBuffer(data)
-	dec := gob.NewDecoder(buf)
+	r.updateRepositoryCursor(ctx, repos)
 
-	cursorData := make(map[string]interface{})
+	return repos, nil
+}
 
-	if err := dec.Decode(&cursorData); err != nil {
-		return err
-	}
+func (r *reminder) updateRepositoryCursor(ctx context.Context, repos []db.Repository) {
+	logger := zerolog.Ctx(ctx)
 
-	if val, ok := cursorData["projectListCursor"]; ok {
-		if v, ok := val.(string); ok {
-			r.projectListCursor = v
-		} else {
-			return fmt.Errorf("projectListCursor is not a string")
-		}
+	if len(repos) == 0 {
+		r.repositoryCursor = uuid.Nil
+	} else {
+		r.repositoryCursor = repos[len(repos)-1].ID
+		r.adjustCursorForEndOfList(ctx)
 	}
 
-	if val, ok := cursorData["repoListCursor"]; ok {
-		if v, ok := val.(map[projectProviderPair]string); ok {
-			r.repoListCursor = v
-		} else {
-			return fmt.Errorf("repoListCursor is not a map[projectProviderPair]string")
-		}
-	}
-
-	return nil
+	logger.Debug().Msgf("updated repository cursor to: %s", r.repositoryCursor)
 }
 
-// TODO: Will be implemented in a separate PR
-func (_ *reminder) sendReminders(_ context.Context) []error {
-	return nil
+func (r *reminder) adjustCursorForEndOfList(ctx context.Context) {
+	logger := zerolog.Ctx(ctx)
+	// Check if the cursor is the last element in the db
+	exists, err := r.store.RepositoryExistsAfterID(ctx, r.repositoryCursor)
+	if err != nil {
+		logger.Error().Err(err).Msgf("unable to check if repository exists after cursor: %s", r.repositoryCursor)
+		logger.Info().Msg("resetting cursor to zero uuid")
```

Comment:
I'd tend to put this as one message in the logs, rather than two separate messages.  Especially when searching structured logs (say, for the last 48 hours across 8 or 10 servers), it can be hard to "scroll" forward or backward to the next log line from a particular server.